### DISCUSSION TITLE
请问是否每次新插入数据在查询之前都需要load，这次load会重新加载所有数据和索引么

### DISCUSSION THREAD

**Discussion by kkpssr** [milvus-io/milvus#30886] [Q&A and General discussion]
如题

**Comment by yhmo** [comment]
不需要，只要调用过一次collection.load()，之后插入的数据都会自动加载到内存里，新的数据如果建好了索引，索引也会自动加载到内存里。

**Reply by secsilm** [reply]
请问每次插入数据后都需要创建一次索引吗？

**Reply by xiaofan-luan** [reply]
不需要，load和index都只需要调用一次

**Reply by secsilm** [reply]
那新数据进来后，多久会加入到索引中呢？

**Reply by yhmo** [reply]
关于数据怎么处理，看这个文章吧。几句话难以讲清楚
https://milvus.io/blog/deep-dive-3-data-processing.md

**Comment by yhmo** [comment]
请各位提问时尽量使用英语，Milvus社区里有世界各地的工程师，用英文问答所有人都能看得懂。
